### PR TITLE
docs: add CREATE TYPE to syntax reference (DOCS-2885)

### PR DESCRIPTION
**Pull request by JimGalasyn**

### CODE REVIEW

File: docs/developer-guide/syntax-reference.rst [hunk 1]
```diff
@@ -788,6 +788,77 @@ The WITH clause supports the following properties:
...
+CREATE TYPE statement works in interactive and headless modes.
+
+Any attempt to register the same type twice, without a corresponding DROP TYPE
+statement, will fail.
+
+**Example**
+
+.. code:: sql
+
+    CREATE TYPE ADDRESS AS STRUCT<number INTEGER, street VARCHAR, city VARCHAR>;
```
Comment (vcrfxia) [inline docs/developer-guide/syntax-reference.rst]: Should we follow up this example with an example usage of the custom type, in order to drive home the point? Not sure if the syntax reference is the best place for that though.

Comment (agavra) [reply docs/developer-guide/syntax-reference.rst]: +1 the example I've been using is to then define:
```
CREATE TYPE PERSON AS STRUCT<name VARCHAR, address ADDRESS>;
```

File: docs/developer-guide/syntax-reference.rst [hunk 2]
```diff
@@ -788,6 +788,77 @@ The WITH clause supports the following properties:
...
+**Synopsis**
+
+.. code:: sql
+
+    DROP TYPE <type_name>;
+
+**Description**
+
+The DROP TYPE statement removes a type alias from KSQL. This statement fails
+if the type is being used in any active query or user-defined function.
```
Comment (agavra) [inline docs/developer-guide/syntax-reference.rst]: I believe the second sentence is no longer true - it doesn't track whether or not queries are using the type, you can drop it at anytime (old queries will still work and will not change if you register a new one)

Comment (JimGalasyn) [reply docs/developer-guide/syntax-reference.rst]: @agavra, I updated with your comment, does it look correct?

Comment (agavra) [reply docs/developer-guide/syntax-reference.rst]: yup!

**Pull request merged**